webapp-django/crashstats/symbols/views.py

```diff
@@ -32,6 +36,46 @@ def check_symbols_archive_content(content, header_length=2):
                 )
 
 
+def unpack_and_upload(iterator):
+    necessary_setting_keys = (
+        'AWS_ACCESS_KEY',
+        'AWS_SECRET_ACCESS_KEY',
+        'SYMBOLS_BUCKET_LOCATION',
+        'SYMBOLS_BUCKET_NAME',
+        'SYMBOLS_FILE_PREFIX',
+    )
+    for key in necessary_setting_keys:
+        if not getattr(settings, key):
+            raise ImproperlyConfigured(
+                "Setting %s must be set" % key
+            )
+
+    conn = boto.connect_s3(
+        settings.AWS_ACCESS_KEY,
+        settings.AWS_SECRET_ACCESS_KEY
+    )
+
+    try:
+        bucket = conn.get_bucket(settings.SYMBOLS_BUCKET_NAME)
+    except boto.exception.S3ResponseError:
+        bucket = conn.create_bucket(
+            settings.SYMBOLS_BUCKET_NAME,
+            settings.SYMBOLS_BUCKET_LOCATION
+        )
+
+    total_uploaded = 0
+    for member in iterator:
+        key = bucket.new_key(os.path.join(
```

Comment:
We expect to see a lot of attempts to upload symbols that are already present in S3. We should do a HEAD here and compare the size of the remote and local files, and not bother to upload if it's already present (```key.get_key()``` can do this, see http://boto.readthedocs.org/en/latest/ref/s3.html#boto.s3.bucket.Bucket.get_key)